cmake/Modules/FindSWIG.cmake

```diff
@@ -0,0 +1,97 @@
+#   SWIG_EXECUTABLE - the path to the swig executable
+#   SWIG_VERSION   - the version number of the swig executable
+#
+#
+#
+# The minimum required version of SWIG can be specified using the
+# standard syntax, e.g.  find_package(SWIG 1.1)
+#
+# All information is collected from the SWIG_EXECUTABLE so the version
+# to be found can be changed from the command line by means of setting
+# SWIG_EXECUTABLE
+
+find_program(SWIG_EXECUTABLE NAMES swig3.0 swig2.0 swig)
+
+add_executable(swig IMPORTED)
+
+if(SWIG_EXECUTABLE)
+  execute_process(COMMAND ${SWIG_EXECUTABLE} -swiglib
+    OUTPUT_VARIABLE SWIG_swiglib_output
+    ERROR_VARIABLE SWIG_swiglib_error
+    RESULT_VARIABLE SWIG_swiglib_result)
+
+  if(SWIG_swiglib_result)
+    if(SWIG_FIND_REQUIRED)
+      message(SEND_ERROR "Command \"${SWIG_EXECUTABLE} -swiglib\" failed with output:\n${SWIG_swiglib_error}")
+    else()
+      message(STATUS "Command \"${SWIG_EXECUTABLE} -swiglib\" failed with output:\n${SWIG_swiglib_error}")
+    endif()
+  else()
+    string(REGEX REPLACE "[\n\r]+" ";" SWIG_swiglib_output ${SWIG_swiglib_output})
+    find_path(SWIG_DIR swig.swg PATHS ${SWIG_swiglib_output} NO_CMAKE_FIND_ROOT_PATH)
+    if(SWIG_DIR)
+      set(SWIG_USE_FILE ${CMAKE_CURRENT_LIST_DIR}/UseSWIG.cmake)
+      execute_process(COMMAND ${SWIG_EXECUTABLE} -version
+        OUTPUT_VARIABLE SWIG_version_output
+        ERROR_VARIABLE SWIG_version_output
+        RESULT_VARIABLE SWIG_version_result)
+      if(SWIG_version_result)
+        message(SEND_ERROR "Command \"${SWIG_EXECUTABLE} -version\" failed with output:\n${SWIG_version_output}")
+      else()
+        string(REGEX REPLACE ".*SWIG Version[^0-9.]*\([0-9.]+\).*" "\\1"
+          SWIG_version_output "${SWIG_version_output}")
+        set(SWIG_VERSION ${SWIG_version_output} CACHE STRING "Swig version" FORCE)
+      endif()
+    endif()
+  endif()
+else()
+  mark_as_advanced(SWIG_EXECUTABLE)
+
+  set(URL https://github.com/swig/swig)
+  set(VERSION fbeb566014a1d320df972aef965daf042db7db36) # Tag origin/rel-3.0.12
+  set_target_description(swig "Simplified Wrapper and Interface Generator (SWIG)" ${URL} ${VERSION})
+  
+  ExternalProject_Add(swig_swig
+      GIT_REPOSITORY ${URL}
+      GIT_TAG        ${VERSION}
+      GIT_PROGRESS   ON
+      # GIT_SHALLOW    ON
+      PATCH_COMMAND ${GIT_EXECUTABLE} apply ${PROJECT_SOURCE_DIR}/patch/add-nodejs8-support-to-swig.patch || true
+      # We should install SWIG to properly access SWIG lib
+      CONFIGURE_COMMAND ./autogen.sh COMMAND ./configure --without-pcre --prefix=${EP_PREFIX}/src/swig_swig
+      BUILD_IN_SOURCE ON
+      BUILD_COMMAND $(MAKE)
+      TEST_COMMAND "" # remove test step
+      )
+  ExternalProject_Get_Property(swig_swig source_dir binary_dir)
+
+  # Predefined vars for local installed SWIG
+  set(SWIG_EXECUTABLE ${binary_dir}/swig)
+  set(SWIG_DIR ${EP_PREFIX}/src/swig_swig)
+  set(SWIG_VERSION 3.0.12)
+  set(SWIG_USE_FILE ${CMAKE_CURRENT_LIST_DIR}/UseSWIG.cmake)
```

Comment:
Not really sure that there's no var for module path, but at least it's possible to use `${CMAKE_ROOT}/Modules/UseSWIG.cmake`

Reply:
Thanks! I totally newbie in CMake, so I added entire UseSWIG.cmake to the project :(